src/styles/_catalog.scss

```diff
@@ -0,0 +1,20 @@
+@use 'variables';
+
+.catalog {
+  padding: 50px 40px;
+  margin: 0;
+
+  display: grid;
+  justify-content: center;
+  gap: 46px 48px;
+
+  @each $width, $columns in variables.$screenWidth {
```

Comment:
Ensure that `variables.$screenWidth` is defined in your 'variables' file. It should contain the necessary breakpoints and corresponding column counts for the responsive grid layout.